### DISCUSSION TITLE
New release (v1.43.0) ?

### DISCUSSION THREAD

**Discussion by natefinch** [golangci/golangci-lint#2273] [Q&A] [answered]
I'd love to get `--whole-files` into a release so I can use it more easily, and it seems like there hasn't been a release in almost a month... maybe now's a good time? :) 🙏🏻

**Comment by ldez** [comment]
Hello,

Usually, we recreate a release when there are enough features (not really based on time). 

I will check the changelog.

**Reply by ldez** [reply]
The current changelog (v1.42.1..master):

* Add tenv linter
* Add nilnil linter
* new-linter: ireturn (checks for function return type)
* new-from-rev: add support for finding issues in entire files in a diff
* nlreturn: add block-size option
* gci: Parse the settings more similarly to the cli
* errcheck: update to HEAD
* bump github.com/tetafro/godot from 1.4.9 to 1.4.10
* bump github.com/Antonboom/errname from 0.1.4 to 0.1.5
* bump github.com/tetafro/godot from 1.4.10 to 1.4.11
* bump github.com/kunwardeep/paralleltest from 1.0.2 to 1.0.3
* bump github.com/butuzov/ireturn from 0.1.0 to 0.1.1

In summary: 3 new linters, 6 linters patch, 2 small features

**Reply by timkral** [reply]
@ldez Picking up this thread again. Looking at the changelog, there are 3 new linters still, but looks like upwards of 30 dependency upgrades since the last release. This looks somewhat on par with the amount of change that triggered the last two releases:

https://github.com/golangci/golangci-lint/compare/v1.40.1...v1.41.0
https://github.com/golangci/golangci-lint/compare/v1.41.1...v1.42.0

VERSUS

https://github.com/golangci/golangci-lint/compare/v1.42.1...master

What's your typical threshold for feature improvement to qualify for a release?

For my part, I'm particularly interested in the Ruleguard updates introduced [here](https://github.com/golangci/golangci-lint/pull/2041)

Thanks!
-Tim

**Reply by ldez** [reply]
The current changelog (v1.42.1..master):

* Add contextcheck linter
* Add nilnil linter
* Add tenv linter
* new-linter: ireturn (checks for function retun)
* errcheck: empty selector name.
* new-from-rev: add support for finding issues in entire files in a diff
* nlreturn: add block-size option
* bump github.com/Antonboom/errname from 0.1.4 to 0.1.5
* bump github.com/butuzov/ireturn from 0.1.0 to 0.1.1
* bump github.com/charithe/durationcheck from 0.0.8 to 0.0.9
* bump github.com/esimonov/ifshort from 1.0.2 to 1.0.3
* bump github.com/go-critic/go-critic from 0.5.6 to 0.6.0
* bump github.com/kunwardeep/paralleltest from 1.0.2 to 1.0.3
* bump github.com/mgechev/revive from 1.1.1 to 1.1.2
* bump github.com/securego/gosec/v2 from 2.8.1 to 2.9.1
* bump github.com/tetafro/godot from 1.4.10 to 1.4.11
* bump github.com/tetafro/godot from 1.4.9 to 1.4.10
* bump github.com/tomarrell/wrapcheck/v2 from 2.3.0 to 2.3.1


In summary: 4 new linters, 10 linters patch, 3 small features

**Reply by timkral** [reply]
Hi @ldez , thank you for that information. Since you make release decisions based on added features, not timing as you stated, I'm wondering how this list compares to the other releases that you've made in the past. Is it enough to warrant a new release? Or will you be waiting for more features to be added before proceeding?

I'm not sure if this matters or not, but the documentation site is already advertising the new release without it being available.

<img width="1372" alt="Screen Shot 2021-10-27 at 9 03 07 PM" src="https://user-images.githubusercontent.com/4274030/139185642-b733a01a-a374-4f9f-8ca6-bd1425955876.png">
<img width="1372" alt="Screen Shot 2021-10-27 at 9 03 35 PM" src="https://user-images.githubusercontent.com/4274030/139185664-13e4e788-7685-4e69-8d34-fc44ea304baf.png">

**Reply by abursavich** [reply]
FWIW, I would also appreciate a new release. My tests run against several versions of go and they've been failing against tip.

```
18.04s$ golangci-lint run
dynamictls.go:21:2: could not import sync/atomic (-: could not load export data: cannot import "sync/atomic" (unknown iexport format version 2), export data is newer version - update tool) (typecheck)
	"sync/atomic"
	^
The command "golangci-lint run" exited with 1.
```

I've tracked it down to needing an update to `golang.org/x/tools`, which is already in master.

**Comment by SVilgelm** [comment]
here is the current changelog

6 new linters, 15 linters patch, 4 patch on linter inside golangci-lint, 2 minor fixes/features

```
* Add github.com/sylvia7788/contextcheck linter
* Add github.com/breml/bidichk linter
* Add github.com/butuzov/ireturn linter
* Add github.com/Antonboom/nilnil linter
* Add github.com/sivchari/tenv linter
* Add github.com/blizzy78/varnamelen linter

* bump 4d63.com/gochecknoglobals to v0.1.0
* bump github.com/Antonboom/errname from 0.1.4 to 0.1.5
* bump github.com/butuzov/ireturn from 0.1.0 to 0.1.1
* bump github.com/charithe/durationcheck from 0.0.8 to 0.0.9
* bump github.com/esimonov/ifshort from 1.0.2 to 1.0.3
* bump github.com/go-critic/go-critic from 0.5.6 to 0.6.0
* bump github.com/jingyugao/rowserrcheck from 1.1.0 to 1.1.1
* bump github.com/kunwardeep/paralleltest from 1.0.2 to 1.0.3
* bump github.com/mgechev/revive from 1.1.1 to 1.1.2
* bump github.com/nakabonne/nestif from 0.3.0 to 0.3.1
* bump github.com/securego/gosec/v2 from 2.8.1 to 2.9.1
* bump github.com/tetafro/godot from 1.4.10 to 1.4.11
* bump github.com/tetafro/godot from 1.4.9 to 1.4.10
* bump github.com/tomarrell/wrapcheck/v2 from 2.3.0 to 2.3.1
* bump github.com/tomarrell/wrapcheck/v2 from 2.3.1 to 2.4.0

* errcheck: empty selector name.
* gocritic: add support for variable substitution in ruleguard path settings
* gosec: filter issues according to the severity and confidence
* nlreturn: add block-size option

* new-from-rev: add support for finding issues in entire files in a diff
* cli: don't hide enable-all option
```

**Comment by ldez** [comment, marked as answer]
The new release (v1.43.0) is available. https://github.com/golangci/golangci-lint/releases/tag/v1.43.0
